src/Core/Webservice/WebserviceCanBeEnabledConfigurationChecker.php

```diff
@@ -0,0 +1,169 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Webservice;
+
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Looks at server configuration in order to check PrestaShop Webservice can be enabled
+ */
+class WebserviceCanBeEnabledConfigurationChecker
```

Comment:
How about `WebserviceConfigurationValidator`with `isValid` and `getErrors` public functions?

Reply:
Also make it final: this will be "our" contract.

Reply:
Are you thinking about a behavior similar to ThemeValidator ?

I'm OK with the idea but the implementation of the ThemeValidator is stateful: https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Addon/Theme/ThemeValidator.php#L50 `getErrors()` must be called **after** `isValid()`.

It is usually recommended to use stateless classes. Wdyt ?

Reply:
Agree with you, maybe only have getErrors() then which is an empty array if it's valid?